app/controllers/users.js

```diff
@@ -1,45 +1,14 @@
-const emailDomain = '@wolox.com.ar',
-  bcrypt = require('bcrypt'),
-  { databaseError, hashError, validationError } = require('../errors'),
+const { validationError } = require('../errors'),
   Users = require('../models').user,
   logger = require('../logger'),
-  saltRounds = 10,
-  minPasswordLength = 8,
-  alphanumericRegex = /^[0-9a-zA-Z]+$/;
-
-const checkValidPassword = pass => pass && pass.length >= minPasswordLength && pass.match(alphanumericRegex);
-
-const checkValidEmail = email => email && email.endsWith(emailDomain);
-
-const hashPassword = pass => bcrypt.hash(pass, saltRounds).catch(error => hashError(error.message));
-
-const createUser = (firstName, lastName, email, password) =>
-  Users.findOrCreate({
-    where: { email },
-    defaults: {
-      firstName,
-      lastName,
-      password
-    }
-  }).catch(error => databaseError(error.message));
+  { hashPassword } = require('../helpers/hasher');
 
 exports.signUp = (req, res, next) => {
   const { first_name: firstName, last_name: lastName, email, password } = req.body;
-  logger.info(`Creating user ${firstName}`);
-
-  if (!firstName || !lastName) {
-    return next(validationError('Missing parameters. firstName and lastName are required'));
-  }
+  logger.info(`Email and password correctly validated. Creating user ${firstName}`);
```

Comment:
`Email and password correctly validated` corresponds to the middleware. Move that to the middelware and keep `Creating user ${firstName}`. Usually the more logs the better.